src/app/public/testing/filter/filter-fixture.spec.ts

```diff
@@ -0,0 +1,155 @@
+   * the SKY UX modules and components in your application for testing convenience. If this has
+   * an adverse effect on your test performance, you can individually bring in each of your app
+   * components and the SKY UX modules that those components rely upon.
+   */
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [FilterTestComponent],
+      imports: [SkyFilterTestingModule]
+    });
+    fixture = TestBed.createComponent(FilterTestComponent);
+    testComponent = fixture.componentInstance;
+    filterSummaryFixture = new SkyFilterFixtureSummary(fixture, DATA_SKY_ID_SUMMARY);
+    filterButtonFixture = new SkyFilterFixtureButton(fixture, DATA_SKY_ID_BUTTON);
+  });
+
+  describe('Summary items', () => {
+    it('should dismiss only filter item from summary', () => {
+      expect(testComponent.appliedFilters.length).toBe(0);
+      testComponent.applyFilter('example');
+      expect(testComponent.appliedFilters.length).toBeGreaterThan(0);
+      fixture.detectChanges();
+      return filterSummaryFixture.filterCloseClick(0).then(() => {
+        fixture.detectChanges();
+        expect(testComponent.appliedFilters.length).toBe(0);
+      });
+    });
+
+    it('should dismiss nth filter item from summary', () => {
+      expect(testComponent.appliedFilters.length).toBe(0);
+      for (let i = 1; i <= 5; i++) {
+        testComponent.applyFilter(`example ${i}`);
+      }
+      expect(testComponent.appliedFilters.length).toBe(5);
+      fixture.detectChanges();
+      return filterSummaryFixture.filterCloseClick(3).then(() => {
+        fixture.detectChanges();
+        expect(testComponent.appliedFilters.length).toBe(4);
+      });
+    });
+
+    it('should throw error when dismissing non existent filter', () => {
+      expect(testComponent.appliedFilters.length).toBe(0);
+      for (let i = 1; i <= 3; i++) {
+        testComponent.applyFilter(`example ${i}`);
+      }
+      expect(testComponent.appliedFilters.length).toBe(3);
+      fixture.detectChanges();
+      return filterSummaryFixture.filterCloseClick(5).then(() => {
+        expect('Error').toBeTrue();
+      }).catch(e => {
+        expect(e).toBeInstanceOf(Error);
+        expect((e as Error).message)
+          .toBe('Unable to click close for a filter index 5');
+      });
+    });
+  });
+
+  describe('Button', () => {
+    it('should click the button', () => {
+      expect(testComponent.buttonClicked).toBeFalse();
+      filterButtonFixture.clickFilterButton();
+      fixture.detectChanges();
+      expect(testComponent.buttonClicked).toBeTrue();
+    });
+
+    it('should do nothing when the button is not enabled', () => {
+      testComponent.buttonIsDisabled = true;
+      expect(testComponent.buttonClicked).toBeFalse();
+      fixture.detectChanges();
+      return filterButtonFixture.clickFilterButton().then(() => {
+        fixture.detectChanges();
+        expect(testComponent.buttonClicked).toBeFalse();
```

Comment:
Just for future reference, another way you could check for a click without having to store a value in your component is to just spy on the event method.
```
const spy = spyOn(testComponent, 'filterButtonClicked');
expect(spy).toHaveBeenCalled();
```
Just a suggestion, but not holding up this PR.